components/nexus-core/src/main/java/org/sonatype/nexus/internal/SonarExample.java

```diff
@@ -0,0 +1,12 @@
+package org.sonatype.nexus.internal;
+
+/**
+ * Created by jason on 7/9/15.
+ */
+public class SonarExample
+{
+  public SonarExample() {
+    String foo = "foo";
+    foo = foo;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid idempotent operations (like assigning a variable to itself). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=pmd%3AIdempotentOperations)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Dead store to foo in new org.sonatype.nexus.internal.SonarExample() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ADLS_DEAD_LOCAL_STORE)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Self assignment of foo in new org.sonatype.nexus.internal.SonarExample() [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASA_LOCAL_SELF_ASSIGNMENT)